Upload Android benchmark results to OSS benchmark database

huydhn opened the pull request:
This PR adds a job to upload Android benchmark results to the benchmark database.  It transforms the `benchmark_results.json` file slightly to fit into the current schema.  We are going to have a better schema soon https://fburl.com/gdoc/ossgtvte, but landing this first would unblock the work on building the dashboard before the launch.  Updating the schema can be done later.

* The job processes what it finds, so if one model fails, the rest will still be uploaded.
* I will follow up with another PR for iOS later. No need to wait for the TPS metric there, we'll upload what available first.

There are still some TODO pending:

* https://github.com/pytorch/test-infra/pull/5742
* https://github.com/pytorch-labs/pytorch-gha-infra/pull/483

But the structure of the CI job is ready to review.

guangy10 commented on .github/scripts/extract_benchmark_results.py:
```diff
@@ -0,0 +1,248 @@
+    from argparse import ArgumentParser
+
+    parser = ArgumentParser(
+        "extract Android benchmark results from AWS Device Farm artifacts"
```
Shouldn't be Android specific right?

guangy10 commented on .github/scripts/extract_benchmark_results.py:
```diff
@@ -0,0 +1,248 @@
+    for r in benchmark_results:
+        r["deviceInfo"]["device"] = job_name
+
+    # TODO (huydhn): This is the current schema of the database oss_ci_benchmark_v2,
+    # and I'm trying to fit ET benchmark results into it, which is kind of awkward.
+    # However, the schema is going to be updated soon
```
The timestamp will be recorded automatically in the db schema right?

huydhn replied:
Yeah, the timestamp is handled on test-infra GHA side https://github.com/pytorch/test-infra/pull/5742

guangy10 reviewed (approved):
LGTM

huydhn commented on .github/scripts/extract_benchmark_results.py:
```diff
@@ -0,0 +1,248 @@
+            "dtype": (
+                r["benchmarkModel"]["quantization"]
+                if r["benchmarkModel"]["quantization"]
+                else "q8"
```
Yeah, @kirklandsign raises the same concern on the internal diff, I will use a string like `n/a` here as an empty string here means no quantization.

huydhn commented:
The testing looks good https://github.com/pytorch/executorch/actions/runs/11189433899/job/31111425608, I can see the data on the database.

huydhn committed:
Upload Android benchmark results to OSS benchmark database (#5808)

Summary:
This PR adds a job to upload Android benchmark results to the benchmark database.  It transforms the `benchmark_results.json` file slightly to fit into the current schema.  We are going to have a better schema soon https://fburl.com/gdoc/ossgtvte, but landing this first would unblock the work on building the dashboard before the launch.  Updating the schema can be done later.

* The job processes what it finds, so if one model fails, the rest will still be uploaded.
* I will follow up with another PR for iOS later. No need to wait for the TPS metric there, we'll upload what available first.

There are still some TODO pending:

* https://github.com/pytorch/test-infra/pull/5742
* https://github.com/pytorch-labs/pytorch-gha-infra/pull/483

But the structure of the CI job is ready to review.

Pull Request resolved: https://github.com/pytorch/executorch/pull/5808

Reviewed By: guangy10, kirklandsign

Differential Revision: D63869876

Pulled By: huydhn
